runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -656,12 +656,13 @@ MM_CopyForwardScheme::calculateObjectDetailsForCopy(MM_ForwardedHeader* forwarde
 	UDATA actualObjectCopySizeInBytes = 0;
 	UDATA hashcodeOffset = 0;
 
-	if(objectModel->isIndexable(objectModel->getPreservedClass(forwardedHeader))) {
-		UDATA numberOfElements = (UDATA)objectModel->getPreservedIndexableSize(forwardedHeader);
-		UDATA dataSizeInBytes = _extensions->indexableObjectModel.getDataSizeInBytes(clazz, numberOfElements);
-		GC_ArrayletObjectModel::ArrayLayout layout = _extensions->indexableObjectModel.getArrayletLayout(clazz, dataSizeInBytes);
-		*objectCopySizeInBytes = _extensions->indexableObjectModel.getSizeInBytesWithHeader(clazz, layout, numberOfElements);
-		hashcodeOffset = _extensions->indexableObjectModel.getHashcodeOffset(clazz, layout, numberOfElements);
+	if(objectModel->isIndexable(clazz)) {
+		GC_ArrayObjectModel* indexableObjectModel = &_extensions->indexableObjectModel;
+		UDATA numberOfElements = (UDATA)indexableObjectModel->getPreservedIndexableSize(forwardedHeader);
+		UDATA dataSizeInBytes = indexableObjectModel->getDataSizeInBytes(clazz, numberOfElements);
+		GC_ArrayletObjectModel::ArrayLayout layout = indexableObjectModel->getArrayletLayout(clazz, dataSizeInBytes);
+		*objectCopySizeInBytes = indexableObjectModel->getSizeInBytesWithHeader(clazz, layout, numberOfElements);
+		hashcodeOffset = indexableObjectModel->getHashcodeOffset(clazz, layout, numberOfElements);
```

Comment:
Should this be simplified into one method say `calculateObjectSizeAndHashcode()` in ArrayObjectModel? Since they all refer to `indexableObjectModel`

Reply:
yeah, it's reasonable

Reply:
whole this method (calculateObjectDetailsForCopy) for Scavenger is already in ObjectModel(Delegate)
ideally we'd like to share it, but it's not trivial, although there is hope (in future). 

till than we can move components of CopyForward variant or even whole method (calculateObjectDetailsForCopy) if possible (they would differ, by signature)